src/app/games/tafl/TaflEscapeThenPieceThenControlHeuristic.ts

```diff
@@ -4,25 +4,20 @@ import { BoardValue } from 'src/app/jscaip/BoardValue';
 import { Player } from 'src/app/jscaip/Player';
 import { TaflPawn } from './TaflPawn';
 import { TaflState } from './TaflState';
-import { TaflPieceAndControlMinimax, TaflPieceAndControlMinimaxMetrics } from './TaflPieceAndControlMinimax';
-import { TaflNode } from './TaflMinimax';
 import { MGPOptional } from 'src/app/utils/MGPOptional';
-import { GameStatus } from 'src/app/jscaip/GameStatus';
+import { TaflMove } from './TaflMove';
 import { Utils } from 'src/app/utils/utils';
+import { TaflPieceAndControlHeuristic, TaflPieceAndControlHeuristicMetrics } from './TaflPieceAndControlHeuristic';
+import { TaflNode } from './TaflRules';
 
-export class TaflEscapeThenPieceThenControlMinimax extends TaflPieceAndControlMinimax {
+export class TaflEscapeThenPieceThenControlHeuristic<M extends TaflMove, S extends TaflState>
```

Comment:
P-e que pour les multimétrique on est pas obligé de mettre du "Then" ?

Reply:
je pense que le nommage faudrait bien le redéfinir quand on automatisera les multimétriques ? Genre quelle différence entre "and" et "then", quid des poids, etc.

Reply:
Bah, le nom visible quand c'est strictement A puis B je propose  A -> B ou juste A > B
Et si on se met à faire du poids 0.6A + 0.2B ou si on veut pas trop numériser A > B (et du coup A -> B pour le précédent)
Sinon p-e qu'à terme si ça se multiplie faudra leurs donner des noms tout robotiques ou à la limite du surnom de baptême :D 

Pour les noms de classes si on veut distinguer les poids des A -> B, Then au lieu du And clairement, et euuuuh, on pensera à comment nommer les pondérés ? (p-e avec des And et on précise Weighted dans les uns et Multimetrics dans les autres ?)

Reply:
Heuristique 1, Heuristique 2, etc !

Tu résouds celui-ci ?